wgengine/userspace.go

```diff
@@ -88,9 +88,9 @@ type userspaceEngine struct {
 
 	wgLock              sync.Mutex // serializes all wgdev operations; see lock order comment below
 	lastCfgFull         wgcfg.Config
-	lastRouterSig       string // of router.Config
-	lastEngineSigFull   string // of full wireguard config
-	lastEngineSigTrim   string // of trimmed wireguard config
+	lastRouterSig       []byte // of router.Config
```

Comment:
I like the immutable peace of mind of strings over []byte here, not having to worry about whether somebody is illegally retaining references to them and mutating them behind my back, and not having to document the retention rules because it's a string and therefore it must be safe.

Once it's a []byte, it gets more complicated.